src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,212 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+class RegistrationServiceImplTest {
+    private static final int NEGATIVE_AGE = -1;
+    private static final int AGE_0 = 0;
+    private static final int AGE_17 = 17;
+    private static final int AGE_18 = 18;
+    private static final int AGE_30 = 30;
+    private static final String FIVE_SYMBOLS = "qwert";
+    private static final String SIX_SYMBOLS = "qwerty";
+    private static final String TEN_SYMBOLS = "?-qwerty@7";
+    private static final String EXISTING_LOGIN = "here_i_am";
+    private static final String NEW_LOGIN = "new_first";
+    private static final String EMPTY_STRING = "";
+    private static final String SPACE_SYMBOL = " ";
+    private static final String TAB_SYMBOL = "  ";
+    private static final String SIX_SPACES = "      ";
+
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        final User existingInStorage = new User();
+        existingInStorage.setLogin(EXISTING_LOGIN);
+        storageDao.add(existingInStorage);
+        user = new User();
+        user.setLogin(NEW_LOGIN);
+        user.setAge(AGE_18);
+        user.setPassword(SIX_SYMBOLS);
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        user = null;
+        try {
+            registrationService.register(user);
+        } catch (RuntimeException e) {
+            assertEquals("Invalid input user", e.getMessage());
+            return;
+        }
```

Comment:
https://howtodoinjava.com/junit5/expected-exception-example/
the same for other try catch

Reply:
Thumbs up!